### PR TITLE
Add Contour 1.20.2

### LINKED ISSUE #4324: Contour package field to enable ipv6 (for Contour xDS server listeners/addresses and Envoy listeners/addresses)
## Feature Request
The Contour package should enable users to opt in to deploying Contour with ipv6 cluster compatible flags for listener addresses etc. This includes addresses for Contour and Envoy (e.g. flags like `--xds-address` and `--envoy-service-http-address` in the Contour Deployment). Ideally this would be a single value to configure, e.g. `contour.listenerIpFamily` or something similar.

## Describe alternatives you've considered
- Make users manually set flags for listen addresses etc.? Not really a good experience

## Additional context
The [cluster configuration](https://projectcontour.io/docs/v1.20.1/configuration/#cluster-configuration) has a field for lookup family used for ExternalName services, we may have to add a caveat to the documentation that if you're using an ipv6 only cluster, you should make sure to set this appropriately as well. It is a lone field in the embedded YAML of the values, so not sure if we can override it?

cc @vmware-tanzu/contour-owners @vmware-tanzu/pkg-contour-owners

### PR DESCRIPTION
**Pull request by sunjayBhatia**
## What this PR does / why we need it
Add package for newly release Contour 1.20.2

See release notes: https://github.com/projectcontour/contour/releases/tag/v1.20.2

Also adds ipv6 compatibility via setting listen addresses to ipv6 for Contour/Envoy since Contour/Envoy are set up to
listen on ipv6 with ipv4 compatibility, see: https://github.com/projectcontour/contour/pull/4584

## Which issue(s) this PR fixes
Fixes: #4324 
(actually makes obsolete, since we don't need a package flag)

## Describe testing done for PR
locally in cluster and package unit tests

## Special notes for your reviewer
N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Still TODO: Haven't pushed the package bundle yet, but would like to get validation on the content first (esp. the ipv6 compatibility part)

**Comment by sunjayBhatia** [comment]
Alternative to https://github.com/vmware-tanzu/community-edition/pull/4594, can be used as an alternate approach

**Comment by sunjayBhatia** [comment]
This maybe needs some thought/consideration, I haven’t been able to break it/tell anything is wrong really, stats listeners/health listeners all work as well as the envoy http/s listeners 👀

**Review by seemiller — CHANGES_REQUESTED** [review]
Please delete the readme from the docs folder.

I've pushed the package to projects-stg.registry.vmware.com and created a package repository for testing:
```
projects-stg.registry.vmware.com/tce/main@sha256:6290251b0bf919e94b18c4d94ecebed975e82a943caa52f1be4090c7129710a4
```

**Review by skriss — CHANGES_REQUESTED** [review]
One little nit, otherwise looks good to me, would be good to get @youngnick's thoughts on the ipv6 stuff too though.

File: addons/packages/contour/1.20.2/package.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  version: 1.20.2
+  releaseNotes: "contour 1.20.2 https://github.com/projectcontour/contour/releases/tag/v1.20.2"
+  releasedAt: 2022-02-24T00:00:00Z
+  licenses:
+    - "Apache 2.0"
+  template:
+    spec:
+      fetch:
+        - imgpkgBundle:
+            image: projects.registry.vmware.com/tce/contour@sha256:0128c3458d57cd637c198b3a6377bb1eca6e23301f27710ac29d9d0ee1c5bd09
```
[the remark this answers is not in the document]
Comment (sunjayBhatia) [reply addons/packages/contour/1.20.2/package.yaml]: updated

**Review by skriss — DISMISSED** [review]

**Review by skriss — APPROVED** [review]
LGTM, I guess the one downside to the latest refactor is that if we add any new cmd-line args upstream, we'll have to remember to add them here too since we're now replacing them all, but that shouldn't be too common.

**Comment by skriss** [comment]
I think it's fine since we really shouldn't be adding more CLI flags, just something to try to remember on the off chance we do.

**Comment by sunjayBhatia** [comment]
@seemiller this is ready for another review when you get a chance

**Review by seemiller — APPROVED** [review]

**Pull request merged**